### PR TITLE
[Merged by Bors] - feat(CategoryTheory/GradedObject): the associator for the composition of bifunctors

### PR DESCRIPTION
**Pull request by joelriou**
When bifunctors `F₁₂`, `G`, `F` and `G₂₃` are equipped with a natural isomorphism `G(F₁₂(X₁, X₂), X₃) ≅ F(X₁, G₂₃(X₂, X₃))` in three variables, then under reasonable assumptions and conditions on indices, there is an associator isomorphism for the action of these bifunctors on graded objects `mapBifunctorMapObj G ρ₁₂.q (mapBifunctorMapObj F₁₂ ρ₁₂.p X₁ X₂) X₃ ≅mapBifunctorMapObj F ρ₂₃.q X₁ (mapBifunctorMapObj G₂₃ ρ₂₃.p X₂ X₃)`.

This shall be used in order to construct the associator isomorphism for the monoidal category structure on graded objects.

---
- [x] depends on: #10667

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/GradedObject/Associator.lean [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.GradedObject.Trifunctor
```
Comment (TwoFX) [inline Mathlib/CategoryTheory/GradedObject/Associator.lean:6]:
```suggestion
import Mathlib.CategoryTheory.GradedObject.Trifunctor

```

File: Mathlib/CategoryTheory/GradedObject/Associator.lean [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+satisfy suitable assumptions, we construct an isomorphism
+`mapBifunctorAssociator associator ρ₁₂ ρ₂₃ X₁ X₂ X₃` between
+`mapBifunctorMapObj G ρ₁₂.q (mapBifunctorMapObj F₁₂ ρ₁₂.p X₁ X₂) X₃` and
+`mapBifunctorMapObj F ρ₂₃.q X₁ (mapBifunctorMapObj G₂₃ ρ₂₃.p X₂ X₃)` in the category
+`GradedObject J C₄`.
+
+This construction shall be used in the definition of the monoidal category structure
+on graded objects indexed by an additive monoid.
+
+-/
```
Comment (TwoFX) [inline Mathlib/CategoryTheory/GradedObject/Associator.lean:26]:
```suggestion
-/

```

**Review by TwoFX — APPROVED** [review]
Thanks!
bors d+

**Comment by joelriou** [comment]
Thanks for the review!

bors merge

**Pull request closed without merging**